Userland/Games/Chess/main.cpp

```diff
@@ -40,6 +46,7 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
 
     TRY(Core::System::unveil("/res", "r"));
     TRY(Core::System::unveil("/bin/ChessEngine", "x"));
+    TRY(Core::System::unveil("/usr/local/bin/stockfish", "x"));
```

Comment:
This will fail if with no such file or directory, if stockfish is not installed or at this path

Reply:
It should work if you also pass it a `c` even though that looks very ugly. Otherwise you need to give it `rx` access to all of `/usr/local/bin`, I think that always exists?

Reply:
>It should work if you also pass it a c even though that looks very ugly.

That's definitely a hack, don't do that.

> Otherwise you need to give it rx access to all of /usr/local/bin, I think that always exists?

Not a fan either.

We can use `Core::DeprecatedFile::resolve_executable_from_environment` and unveil the result.

Reply:
Serves me right for taking shortcuts while testing.  I've tested it properly this time, I swear!

I've used the error code returned from `unveil` to determine if the file exists or not. A similar pattern is used in `SystemMonitor`

Reply:
> We can use Core::DeprecatedFile::resolve_executable_from_environment and unveil the result.

Actually, I think this is nicer. I'll do this.

Reply:
> > We can use Core::DeprecatedFile::resolve_executable_from_environment and unveil the result.
> 
> Actually, I think this is nicer. I'll do this.

OK, so I tried to do this and `resolve_executable_from_environment()` works fine for ChessEngine but doesn't return anything for stockfish. It complains that the path `/usr/local/bin/stockfish` hasn't been unveiled. 

I'm not sure why it works for one and not for the other. Since the way I'm currently doing it seems to work, I'll leave it as it is for now.

Reply:
I've updated this to use `FileSystem::resolve_executable_from_environment()`, as discussed on discord.